comms/dht/migrations/2020-05-14-103121_add_body_hash/up.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE stored_messages
+    ADD body_hash TEXT;
```

Comment:
We can let the database do the unique checking by using a UNIQUE INDEX. 

Can add after adding the column:
`CREATE UNIQUE INDEX uidx_stored_messages_body_hash ON stored_messages (body_hash);`

If a duplicate hash is inserted, an error will be thrown. You'd need to handle that error case in some way.Perhaps we should make the insert_message function idempotent i.e. if the message hash is there already (duplicate key error) , return `Ok(())`, not sure if we want to call the function `insert_message_if_unique` or something - You'll need to check into the details of how diesel handles unique indexes and insert errors (https://docs.diesel.rs/diesel/result/enum.DatabaseErrorKind.html). 

Alternatively, you can add StorageError::UniqueViolation and handle that one level up.

Reply:
Change applied, the database will perform the unique checking and the `insert_message_if_unique` function was made idempotent.